test/integration/noderesourcetopology_cache_test.go

```diff
@@ -0,0 +1,449 @@
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "32", "30"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "64Gi", "62Gi"),
+							noderesourcetopology.MakeTopologyResInfo(nicResourceName, "2", "2"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "32", "30"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "64Gi", "62Gi"),
+							noderesourcetopology.MakeTopologyResInfo(nicResourceName, "2", "2"),
+						}).Obj(),
+			},
+		},
+		{
+			name: "GU pod: pessimistic cache overallocation prevents pod to be scheduled",
+			podDescs: []podDesc{
+				{
+					podName:      "nrt-pod-1000",
+					isGuaranteed: true,
+					resourcesMap: map[string]string{
+						string(corev1.ResourceCPU):    "16",
+						string(corev1.ResourceMemory): "24Gi",
+					},
+					expectedNode: "fake-node-cache-1",
+				},
+				{
+					podName:      "nrt-pod-2000",
+					isGuaranteed: true,
+					resourcesMap: map[string]string{
+						string(corev1.ResourceCPU):    "16",
+						string(corev1.ResourceMemory): "24Gi",
+					},
+					expectedNode: "",
+				},
+			},
+			nodeResourceTopologies: []*topologyv1alpha1.NodeResourceTopology{
+				MakeNRT().Name("fake-node-cache-1").Policy(topologyv1alpha1.SingleNUMANodeContainerLevel).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "32", "30"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "64Gi", "60Gi"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "32", "30"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "64Gi", "62Gi"),
+						}).Obj(),
+				MakeNRT().Name("fake-node-cache-2").Policy(topologyv1alpha1.SingleNUMANodeContainerLevel).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "32", "10"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "64Gi", "14Gi"),
+						}).
+					Zone(
+						topologyv1alpha1.ResourceInfoList{
+							noderesourcetopology.MakeTopologyResInfo(cpu, "32", "8"),
+							noderesourcetopology.MakeTopologyResInfo(memory, "64Gi", "10Gi"),
+						}).Obj(),
+			},
+		},
+		{
+			name: "GU pod: pessimistic cache overallocation ignores deletes prevents pod to be scheduled",
+			podDescs: []podDesc{
+				{
+					podName:      "nrt-pod-3000",
+					isGuaranteed: true,
+					resourcesMap: map[string]string{
+						string(corev1.ResourceCPU):    "16",
+						string(corev1.ResourceMemory): "24Gi",
+					},
+					expectedNode: "fake-node-cache-1",
+				},
+				{
+					podName:      "nrt-pod-3000",
+					isGuaranteed: true,
+					isDelete:     true,
+					resourcesMap: map[string]string{
+						string(corev1.ResourceCPU):    "16",
+						string(corev1.ResourceMemory): "24Gi",
+					},
+					expectedNode: "fake-node-cache-1",
```

Comment:
Seems a bit odd to specify `expectedNode` when this pod is being deleted. Would probably better to check for err here instead?

Reply:
good point. Let me rearrange the code a bit